Sonar Props added

lgriffin opened the pull request:
Props added

lgriffin committed:
Sonar Props added

fheng-build-servers commented:
Build finished. No test results found.

wtrocki commented:
Retest this please

fheng-build-servers commented:
Build finished. No test results found.

fheng-build-servers commented:
Test comment.

wtrocki commented on sonar-project.properties:
```diff
@@ -0,0 +1,10 @@
+sonar.sources=./lib
+sonar.tests=./test
+sonar.language=js
+sonar.javascript.lcov.reportPath=./coverage/lcov.info
```
Sonar cannot find this file and places comment. We would need to make sure that unit tests and coverage file is regenerated.

lgriffin replied:
General note here: We can of course investigate and make relevant JIRAs one we have the builds consistent. The above config file is a template for the core configuration that should be included so the coverage path could be wrong (again a general note). 

In the above case the file is present it is just empty. There is a cobertura report there however.Something to research further down the line as to why it's not generating or if we can use an alternative.

wtrocki commented:
Retest this please

fheng-build-servers commented:
Build finished. No test results found.

danbev committed:
Adding istanbul code coverage.

Motivation:
Currently there is no code coverage created when unit tests are run.
This commit adds istanbul to correct this.

Modifications:
Updated Gruntfile with istanbul settings and added an exclude for the
cov-unit directory in .gitignore.

Result:
Running the default build will generate a coverage report:
$ grunt
$ open cov-unit/lcov-report/index.html

fheng-build-servers commented:
Build finished. No test results found.

danbev commented:
Retest this please

fheng-build-servers commented:
Build finished. No test results found.

danbev commented:
Retest this please

fheng-build-servers commented:
Build finished. No test results found.

danbev commented:
Retest this please

danbev commented:
@lgriffin I've update your PR with a commit to include the coverage. 

I also added the engineering team which was missing from the fh-logger github repository. This team includes`fheng-build-servers`` which I believe was the issue with the communication between Jenkins and the fh-logger github repo. Can you take a look and if all looks alright I'll merge this.

lgriffin commented:
Hey @danbev thanks for checking this out -- we thought it was a global permissions issue but that makes sense (it might have resolved another issue for us as well!). 

The Sonar props look okay from my perspective, if you wish to merge it I can run a manual test ahead of tonight and make any tweaks necessary.

lgriffin commented:
Analysis successfully ran, including line coverage report.
